### PR TITLE
Fix logserver

### PR DESCRIPTION
**Pull request by yyongycy**
Fix logserver

### CODE REVIEW

**Review by dbw9580 — APPROVED** [review]
LGTM, thanks!

**Comment by yyongycy** [comment]
The following failure has no relation to this change:


Error: 1.651 [ERROR] Failures: 
Error: 1.652 [ERROR] alluxio.master.file.replication.ReplicationCheckerTest.heartbeatFileUnderReplicatedBy1
Error: 1.653 [ERROR]   Run 1: ReplicationCheckerTest.heartbeatFileUnderReplicatedBy1:325 expected:<{16777216=1}> but was:<{}>
Error: 1.653 [ERROR]   Run 2: ReplicationCheckerTest.heartbeatFileUnderReplicatedBy1:325 expected:<{16777216=1}> but was:<{}>
14:46:31.653 [INFO] 
Error: 1.653 [ERROR] alluxio.master.file.replication.ReplicationCheckerTest.heartbeatFileUnderReplicatedBy10
Error: 1.653 [ERROR]   Run 1: ReplicationCheckerTest.heartbeatFileUnderReplicatedBy10:362 expected:<{16777216=10}> but was:<{}>
Error: 1.654 [ERROR]   Run 2: ReplicationCheckerTest.heartbeatFileUnderReplicatedBy10:362 expected:<{16777216=10}> but was:<{}>
14:46:31.654 [INFO] 
Error: 1.654 [ERROR] alluxio.master.file.replication.ReplicationCheckerTest.heartbeatMultipleFilesUnderReplicated
Error: 1.654 [ERROR]   Run 1: ReplicationCheckerTest.heartbeatMultipleFilesUnderReplicated:375 expected:<{16777216=1, 33554432=2}> but was:<{}>
Error: 1.654 [ERROR]   Run 2: ReplicationCheckerTest.heartbeatMultipleFilesUnderReplicated:375 expected:<{16777216=1, 33554432=2}> but was:<{}>

**Review by yuzhu — APPROVED** [review]

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

File: logserver/src/main/java/alluxio/logserver/AlluxioLog4jSocketNode.java [hunk 1]
```diff
@@ -50,6 +49,15 @@
   private final String mBaseLogsDir;
   private final Socket mSocket;
 
+  private static void setAcceptList(ValidatingObjectInputStream validatingObjectInputStream) {
+    validatingObjectInputStream.accept(java.util.Hashtable.class);
+    validatingObjectInputStream.accept("java.lang.*", "[Ljava.lang.*");
+    validatingObjectInputStream.accept(org.apache.log4j.spi.LoggingEvent.class);
```
Comment (Xenorith) [inline logserver/src/main/java/alluxio/logserver/AlluxioLog4jSocketNode.java:55]: how would we know if there is a new class in a newer version of log4j that we would need to add to the accept list? is it possible to write a unit test?

if we cannot write a unit test to ensure forwards compatibility, are there better ways than to list out the specific classes? for example, is it safe to include everything in the org.apache.log4j.spi package? https://logging.apache.org/log4j/2.x/log4j-1.2-api/apidocs/org/apache/log4j/spi/package-tree.html

Comment (yyongycy) [reply logserver/src/main/java/alluxio/logserver/AlluxioLog4jSocketNode.java:55]: This is something like the white list, and white list is requiring to add that per change.
Yes, an ut can be written to protect that.
Create an issue for this: https://github.com/Alluxio/alluxio/issues/14767

**Comment by Xenorith** [comment]
alluxio-bot, cherry-pick this to branch-2.7 please

**Pull request merged**